### PR TITLE
Utilize `ApplicationConfig["API_PER_PAGE_MAX"]`

### PR DESCRIPTION
**Pull request by andygeorge**
## What type of PR is this? (check all applicable)

- [x] Feature
- [x] Bug Fix

## Description

This adds support for using `ApplicationConfig["API_PER_PAGE_MAX"]` within the API. When set to an integer value in `rails.env`, API calls will be limited to a `per_page` value no greater than `API_PER_PAGE_MAX`.

For example, if this is set in `rails.env`:
```
API_PER_PAGE_MAX=10
```

...the following request will be limited to 10 results per-page:
```
/api/articles?per_page=999
```

If `API_PER_PAGE_MAX` is unset, the per-page max functions as it does already: generally limited to 1000, with a few exceptions.

## Related Tickets & Documents

https://github.com/forem/infra/issues/509

### CODE REVIEW

**Comment by andygeorge** [comment]
Testing this on Forem Cloud.

**Comment by andygeorge** [comment]
Tested successfully:

- Canary1, correctly limited to 10 results
```
irb(main):001:0> ENV["API_PER_PAGE_MAX"]
=> "10"
```
- Canary2, correctly "unlimited" (well, 1000) results
```
irb(main):001:0> ENV["API_PER_PAGE_MAX"]
=> nil
```

File: app/controllers/concerns/api/organizations_controller.rb [hunk 1]
```diff
@@ -20,6 +20,9 @@ module OrganizationsController
     ].freeze
     private_constant :LISTINGS_FOR_SERIALIZATION
 
+    PER_PAGE_MAX = (ENV["API_PER_PAGE_MAX"] || 1000).to_i
```
Comment (fdocr) [inline app/controllers/concerns/api/organizations_controller.rb:23]: Same as above

```suggestion
    PER_PAGE_MAX = (ApplicationConfig["API_PER_PAGE_MAX"] || 1000).to_i
```

File: app/controllers/concerns/api/podcast_episodes_controller.rb [hunk 2]
```diff
@@ -2,10 +2,13 @@ module Api
   module PodcastEpisodesController
     extend ActiveSupport::Concern
 
+    PER_PAGE_MAX = (ENV["API_PER_PAGE_MAX"] || 1000).to_i
```
Comment (fdocr) [inline app/controllers/concerns/api/podcast_episodes_controller.rb:5]: Same as above

```suggestion
    PER_PAGE_MAX = (ApplicationConfig["API_PER_PAGE_MAX"] || 1000).to_i
```

File: app/controllers/concerns/api/tags_controller.rb [hunk 3]
```diff
@@ -5,10 +5,13 @@ module TagsController
     ATTRIBUTES_FOR_SERIALIZATION = %i[id name bg_color_hex text_color_hex].freeze
     private_constant :ATTRIBUTES_FOR_SERIALIZATION
 
+    PER_PAGE_MAX = (ENV["API_PER_PAGE_MAX"] || 1000).to_i
```
Comment (fdocr) [inline app/controllers/concerns/api/tags_controller.rb:8]: Same as above

```suggestion
    PER_PAGE_MAX = (ApplicationConfig["API_PER_PAGE_MAX"] || 1000).to_i
```

File: app/controllers/concerns/api/videos_controller.rb [hunk 4]
```diff
@@ -7,10 +7,13 @@ module VideosController
     ].freeze
     private_constant :INDEX_ATTRIBUTES_FOR_SERIALIZATION
 
+    PER_PAGE_MAX = (ENV["API_PER_PAGE_MAX"] || 1000).to_i
```
Comment (fdocr) [inline app/controllers/concerns/api/videos_controller.rb:10]: Same as above

```suggestion
    PER_PAGE_MAX = (ApplicationConfig["API_PER_PAGE_MAX"] || 1000).to_i
```

File: app/services/article_api_index_service.rb [hunk 5]
```diff
@@ -1,6 +1,6 @@
 class ArticleApiIndexService
-  DEFAULT_PER_PAGE = 30
-  MAX_PER_PAGE = 1000
+  MAX_PER_PAGE = (ENV["API_PER_PAGE_MAX"] || 1000).to_i
```
Comment (fdocr) [inline app/services/article_api_index_service.rb:2]: Same as above

```suggestion
  MAX_PER_PAGE = (ApplicationConfig["API_PER_PAGE_MAX"] || 1000).to_i
```

**Comment by andygeorge** [comment]
@fdocr great call, and yes - I would LOVE some help on tests! I am bad at rspec :joy: .

**Review by joshpuetz — APPROVED** [review]
The only concern I would have is updating API documentation, otherwise looks good to me!

**Review by fdocr — APPROVED** [review]
Sweet, I can submit a PR with the specs + tweaks to the docs after lunch.

I don't want to block this since it looks good and is tested to work on canary instances so approving now 👍🏼

**Comment by andygeorge** [comment]
@joshpuetz good call, I believe I updated the API docs correctly.

@joshpuetz and @fdocr could I get a re-:eyes:? Thank you!

**Pull request merged**